Split TextInput into TextInputOutlined and TextInputFlat

pbitkowski opened the pull request:
What this PR consists of:
- make TextInput proxy component that renders TextInputOutlined or TextInputFlat based on 'mode' props,
- move TextInput components to its own directory and change imports in index to make it work.

### Motivation

Issue #791 

### Test plan

Open example and compare it with version without my changes.

pbitkowski commented:
I have an issue with Flow here. It logs a long list of errors connected to possible usage of TextInput (from React Native) props on our TextInput. I used the same Properties as before so it's weird that it doesn't work. I'm not sure if the intersaction of `React.ElementConfig` and Object exact type is correct in TextInput props. Could you check it @ferrannp ? Thanks :)

satya164 commented:
I think the state can be moved up to the `TextInput` component to reduce duplication. Let's also move the types to a common file. The documentation can be removed from the individual components as well.

pbitkowski commented:
@satya164 I moved types to common file and used spread operator for extending React Native TextInput Props but still got errors, which is weird. BTW. I rebased my branch to master and some TS errors appeared.

pbitkowski commented:
Fixed flow errors and rebased. Could you take a look @Trancever ?

Trancever commented on src/components/TextInput/TextInputOutlined.js:
```diff
@@ -501,24 +315,24 @@ class TextInput extends React.Component<Props, State> {
 
-        {mode === 'outlined' && label ? (
+        {label ? (
           // When mode == 'outlined', the input label stays on top of the outline
```
We can remove `When mode == 'outlined` from comment :D

pbitkowski replied:
Done

pbitkowski commented on src/components/TextInput/TextInputFlat.js:
```diff
@@ -0,0 +1,479 @@
+
+    return (
+      <View style={[containerStyle, style]}>
+        {/* When mode === 'flat', render an underline */}
```
Done

pbitkowski commented on src/components/TextInput/TextInput.js:
```diff
@@ -0,0 +1,97 @@
+    error: false,
+    multiline: false,
+    editable: true,
+    render: props => <NativeTextInput {...props} />,
```
Done
